src/main/java/org/stellar/sdk/MemoId.java

```diff
@@ -1,18 +1,17 @@
 package org.stellar.sdk;
 
+import com.google.common.base.Objects;
 import org.stellar.sdk.xdr.MemoType;
 import org.stellar.sdk.xdr.Uint64;
 
+
 /**
  * Represents MEMO_ID.
  */
 public class MemoId extends Memo {
   private long id;
 
   public MemoId(long id) {
-    if (Long.compareUnsigned(id, 0) < 0) {
-      throw new IllegalArgumentException("id must be a positive number");
-    }
```

Comment:
is it possible for an unsigned long to be less than 0? I thought the if statement will always evaluate to false

Reply:
There is no unsigned `long` type in Java. `Long.compareUnsigned` method is there to treat `long` as it was unsigned (if developers passes an unsigned value created using `Long.parseUnsignedLong`). It's all done because memo ID value is `uint64` in XDR files.

Reply:
OK, you're right! This comes from the fix in for #129. I haven't noticed that actually the whole space of values for `long` is allowed.